### DISCUSSION TITLE
Cannot preview mp4 files

### DISCUSSION THREAD

**Discussion by PresidentialAnt** [coder/code-server#6600] [Q&A]
When I select an mp4 in the explorer, the main display only shows. "An error occured while loading the video file."
The mp4 definitely works, as if I download it I can play it locally, but for some reason not through code-server.

I have even tried installing the extension MPEG-4 Preview, but nothing changed.

Below is a screenshot

![image](https://github.com/coder/code-server/assets/40330039/8b5608db-1bb7-42fa-a125-5fcff6ab8a47)

**Comment by code-asher** [comment]
Are you using a secure context like HTTPS or localhost?  Web views require secure contexts so that might be why.

I experimented with an mp4 and it worked for me (code-server v4.20.0).

**Reply by PresidentialAnt** [reply]
This particular screenshot was taken from localhost, but I typically connect through HTTPS and I have the same issue. 

Out of curiousity what browser do you use?

**Reply by code-asher** [reply]
I used Chromium 120.0.6099.129 on Linux and I was accessing over HTTPS with a valid certificate.

**Reply by code-asher** [reply]
Interesting, it did not work for me in Firefox though.

**Reply by code-asher** [reply]
I was able to reproduce in Codespaces, so it seems to be an upstream issue with VS Code.
